### PR TITLE
Issue #41: Added unit ID to log statements, added some log statements…

### PR DESCRIPTION
**Pull request by boons605**
… to track application progress, ensured setting of connected flag on Xbee modules, added statement on initialization of Xbee network to kick it into a mode where joining is allowed.

### CODE REVIEW

File: CentralUnit/Communication/XbeeNetwork.cs [hunk 1]
```diff
@@ -346,22 +357,33 @@ private void ProcessXbeePacket(XBeePacket packet)
         {
             if (packet != null)
             {
+                if (Log.IsDebugEnabled)
+                {
+                    Log.Debug(packet.ToPrettyString());
```
Comment (CptMalfunction) [inline CentralUnit/Communication/XbeeNetwork.cs:362]: I dont know a lot about this logging library. But i would assume Log.Debug only does anything if debug is enabled right?

File: CentralUnit/Communication/XbeeNetwork.cs [hunk 2]
```diff
@@ -346,22 +357,33 @@ private void ProcessXbeePacket(XBeePacket packet)
...
+                    Log.Info($"Got response for AT command {atrPacket.StringCommandValue}, status {atrPacket.Status}");
+                }
                 else
                 {
-                    Log.Info($"Got unknown packet of type {packet.GetType().Name}");
-                    if (Log.IsDebugEnabled)
-                    {
-                        Log.Debug(packet.ToPrettyString());
-                    }
+                    Log.Info($"Got unknown packet of type {packet.GetType().Name}");
```
Comment (CptMalfunction) [inline CentralUnit/Communication/XbeeNetwork.cs:386]: switch syntax also supports pattern matching like `packet is TransmitStatusPacket txsPacket` so i think thisif/else block is big ewnough to become a switch

File: CentralUnit/Communication/XbeeNetwork.cs (hunk 1 above)
Comment (boons605) [reply CentralUnit/Communication/XbeeNetwork.cs:362]: The logging method only logs when debug is enabled, but that means a string would be generated every time this code is called. 

Adding the if statement saves generating a string.

File: CentralUnit/Communication/XbeeNetwork.cs (hunk 2 above)
Comment (boons605) [reply CentralUnit/Communication/XbeeNetwork.cs:386]: Wasn't aware of this. Will refactor

**Pull request merged**